plugin-common/src/main/java/org/opennms/plugins/cloud/grpc/CloudLogService.java

```diff
@@ -28,20 +28,115 @@
 
 package org.opennms.plugins.cloud.grpc;
 
+import static java.util.Objects.isNull;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.apache.commons.lang3.StringUtils.contains;
+import static org.opennms.plugins.cloud.grpc.CloudLogServiceUtil.convertToLatencyTraceList;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.opennms.integration.api.v1.timeseries.StorageException;
+import org.opennms.plugins.cloud.srv.GrpcService;
+import org.opennms.tsaas.telemetry.GatewayGrpc;
+import org.opennms.tsaas.telemetry.GatewayOuterClass;
+
+import com.google.common.annotations.VisibleForTesting;
+
 import io.grpc.MethodDescriptor;
-import io.grpc.Status.Code;
+import io.grpc.Status;
+import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 
-/**
- * Logs Requests and sends them in batches to the cloud.
- */
 @Slf4j
-public class CloudLogService {
+public class CloudLogService implements GrpcService {
+
+    private static ConcurrentLinkedQueue<LogEntry> logEntryQueue;
```

Comment:
I think this can be not static? it is just a member variable. The CloudLogServiceitself is a Singleton via bean definition, no?